couchdb/server/state.py

```diff
@@ -2,11 +2,37 @@
 #
 import logging
 
-__all__ = ('add_fun', 'add_lib')
+__all__ = ('add_fun', 'add_lib', 'reset')
 
 log = logging.getLogger(__name__)
 
 
+def reset(server, config=None):
+    """Resets query server state.
+
+    :command: reset
+
+    :param server: Query server instance.
+    :type server: :class:`~couchdb.server.BaseQueryServer`
+
+    :param config: Optional dict argument to set up query config.
+    :type config: dict
+
+    :return: True
+    :rtype: bool
+    """
+    log.debug('Reset server state')
+    del server.state['functions'][:]
+    del server.state['functions_src'][:]
+    server.state['query_config'].clear()
+    if config is not None:
+        log.debug('Set new query config:\n%s', config)
+        server.state['query_config'].update(config)
+    if server.version >= (1, 1, 0):
+        server.state['view_lib'] = ''
+    return True
```

Comment:
Well, this option also comes from CouchDB with the reset command (:
http://docs.couchdb.org/en/1.6.1/query-server/protocol.html#reset
So no reason to care about - that's not something we contol.